Packs/MicrosoftECM/Integrations/MicrosoftECM/MicrosoftECM.ps1

```diff
@@ -0,0 +1,1074 @@
+. $PSScriptRoot\CommonServerPowerShell.ps1
+$COLLECTION_TYPE_MAPPING = @{
+	"0" = "Root"
+	"1" = "User"
+	"2" = "Device"
+	"3" = "Unknown"
+}
+$COLLECTION_CURRENT_STATUS_MAPPING = @{
+	"0" = "NONE"
+	"1" = "READY"
+	"2" = "REFRESHING"
+	"3" = "SAVING"
+	"4" = "EVALUATING"
+	"5" = "AWAITING_REFRESH"
+	"6" = "DELETING"
+	"7" = "APPENDING_MEMBER"
+	"8" = "QUERYING"
+}
+$SCRIPT_APPROVAL_STATE = @{
+	"0" = "Waiting for approval"
+	"1" = "Declined"
+	"3" = "Approved"
+}
+<#
+.DESCRIPTION
+This function converts a datetime object onto ISO format string
+
+.PARAMETER date
+The date that should be parsed
+
+.OUTPUTS
+Return The String representation of the datetime object normalized to UTC if or $null if $date is $null
+#>
+Function ParseDateTimeObjectToIso($date)
+{
+	if ($date)
+	{
+		return $date.ToUniversalTime().ToString("yyyy-mm-ddTHH:MM:ssZ")
+	}
+	return $null
+}
+<#
+.DESCRIPTION
+This function Verifies only one of the following arguments was actually given and throws an exception if not.
+
+.PARAMETER $errorMessage
+The error message with which the error should be raised
+
+.PARAMETER $parameters
+The parameters list from which only non-null parameter should be given
+#>
+Function AssertNoMoreThenOneParameterGiven
+{
```

Comment:
`$args.Count`
should do the trick instead of everything written here I think, worth checking as this seems to be overkill

Reply:
https://stackoverflow.com/questions/22490543/powershell-how-to-count-the-number-of-arguments
to iterate how it's used